### PR TITLE
use BSD / GNU agnotic sed params

### LINKED ISSUE #1243: build fails on macOS using GNU sed
On macs that use the GNU tooling over BSD, the build fails with the following error. The date error below seems to be a symptom of the sed issue.

```
% ./build/run make -j4                                                                                     17-11-17 - 11:18:40
==== building the cross container (this could take minutes the first time)
=== cross build image failed for build-f18d7920/cross-amd64
date: invalid option -- 'v'
Try 'date --help' for more information.
date: invalid option -- 'v'
Try 'date --help' for more information.
=== docker build build-f18d7920/rootfs-builder-amd64
sed: can't read s|BASEIMAGE|ubuntu:zesty|g: No such file or directory
make[1]: *** [Makefile:36: rootfs-build] Error 2
make: *** [Makefile:27: base.linux_amd64] Error 2
```

### PR DESCRIPTION
**Pull request by wilreichert**
fixes #1243

### CODE REVIEW

**Comment by bassam** [comment]
what about `date` and the `-v` issue?

File: images/image.mk [hunk 1]
```diff
@@ -127,12 +127,7 @@ PRUNE_KEEP ?= 24
 PRUNE_DRYRUN ?= 0
 
 CACHE_DATE_FORMAT := "%Y-%m-%d.%H%M%S"
-ifeq ($(UNAME_S),Linux)
-CACHE_PRUNE_DATE := $(shell date -u --date="$(PRUNE_HOURS) hours ago" +"$(CACHE_DATE_FORMAT)")
-endif
-ifeq ($(UNAME_S),Darwin)
-CACHE_PRUNE_DATE := $(shell date -u -v -$(PRUNE_HOURS)H +"$(CACHE_DATE_FORMAT)")
-endif
+CACHE_PRUNE_DATE := $(shell export TZ="UTC+$(PRUNE_HOURS)"; date +"$(CACHE_DATE_FORMAT)")
```
Comment (bassam) [inline images/image.mk]: why is it `UTC+$(PRUNE_HOURS)` and not `UTC-$(PRUNE_HOURS)`?

Comment (wilreichert) [reply images/image.mk]: not sure TBH, but it works
```
bash-4.4$ TZ="UTC" date
Sat Nov 18 00:15:12 UTC 2017
bash-4.4$ TZ="UTC+24" date
Fri Nov 17 00:15:17 UTC 2017
```

Comment (bassam) [reply images/image.mk]: odd. from https://www.w3.org/TR/NOTE-datetime:

```Times are expressed in local time, together with a time zone offset in hours and minutes. A time zone offset of "+hh:mm" indicates that the date/time uses a local time zone which is "hh" hours and "mm" minutes ahead of UTC. A time zone offset of "-hh:mm" indicates that the date/time uses a local time zone which is "hh" hours and "mm" minutes behind UTC```

Comment (bassam) [reply images/image.mk]: oh its the reverse. if the timezone its ahead of UTC then the current time is behind. confusing.

**Comment by bassam** [comment]
can you please squash the two commits?

**Pull request merged**